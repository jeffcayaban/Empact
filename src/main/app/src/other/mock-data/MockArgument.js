import {CRITICAL_QUESTION_IDS, EXPERT_OPINION, ROLE_ADMIN} from "../../constants";

export default {
    id: 'testId',
    isSupporting: true,
    rootPetitionId: 'testRootPetitionId',
    parentArgumentId: 'testParentArgumentId',
    subCriticalQuestionId: null,
    agreesWithCQ: true,
    criticalQuestionId: CRITICAL_QUESTION_IDS[0],
    creationDateTime: 1551716401541,
    lastUpdatedDateTime: 1551716402000,
    createdBy: {
        id: 'testId',
        username: 'testUsername',
        firstName: 'testFirstName',
        lastName: 'testLastName',
        authorities: [ROLE_ADMIN]
    },
    argumentType: EXPERT_OPINION,
    isAnonymous: false,
    sources: [],
    expert: "testExpert",
    expertDomain: 'testDomain',
    expertAssertion: 'testAssertion'
};