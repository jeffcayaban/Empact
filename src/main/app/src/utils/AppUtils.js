import {API_BASE_URL, AUTHORIZATION_HEADER_FIELD} from '../constants';
import {ACCESS_TOKEN, AUTHORIZATION_TOKEN_TYPE, PETITION_LIST_SIZE} from "../constants";

/** Creates a request object and sends the request to the server. **/
const request = (options) => {
    const headers = new Headers({ 'Content-Type': 'application/json' });
    const accessToken = localStorage.getItem(ACCESS_TOKEN);

    if (accessToken !== undefined && accessToken !== null) {
        headers.append(AUTHORIZATION_HEADER_FIELD, `${AUTHORIZATION_TOKEN_TYPE} ${accessToken}`)
    }

    options = {...options, headers: headers};

    return fetch(options.url, options).then(response =>
            (response.json()).then(jsonResponse => (!response.ok) ? Promise.reject(jsonResponse) : jsonResponse)
    );
};

// ---------------------------- APIs ----------------------------

/** Gets all the petitions **/
export function getAllPetitions(page, size, sort, showClosed) {
    page = page || 0;
    size = size || PETITION_LIST_SIZE;

    return request({
        url: API_BASE_URL + "/petitions?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Gets a single petition given an ID **/
export function getPetition(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/" + petitionId,
        method: 'GET'
    });
}

/** Creates a new petition given its details **/
export function createPetition(petitionData) {
    return request({
        url: API_BASE_URL + "/petitions/create",
        method: 'POST',
        body: JSON.stringify(petitionData)
    });
}

/** Deletes an existing petition given its details **/
export function deletePetition(deletePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/delete",
        method: 'DELETE',
        body: JSON.stringify(deletePetitionData)
    });
}

/** Updates an existing petition with the given details **/
export function editPetition(editPetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/edit/",
        method: 'POST',
        body: JSON.stringify(editPetitionData)
    });
}

/** Logs in into a user account **/
export function login(loginRequest) {
    return request({
        url: API_BASE_URL + "/auth/signInUser",
        method: 'POST',
        body: JSON.stringify(loginRequest)
    });
}

/** Creates a new user account given the account details **/
export function signup(signupRequest) {
    return request({
        url: API_BASE_URL + "/auth/signUpUser",
        method: 'POST',
        body: JSON.stringify(signupRequest)
    });
}

/** Checks the username's availability given a username **/
export function checkUsernameAvailability(username) {
    return request({
        url: API_BASE_URL + "/user/checkUsernameAvailability?username=" + username,
        method: 'GET'
    });
}

/** Deletes a user **/
export function deleteUser(deleteUserRequest) {
    return request({
        url: API_BASE_URL + "/auth/deleteUser",
        method: 'DELETE',
        body: JSON.stringify(deleteUserRequest)
    })
}

/** Fetches the current user **/
export function getCurrentUser() {
    if(!localStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }

    return request({
        url: API_BASE_URL + "/user/getCurrentUser",
        method: 'GET'
    });
}

/** Fetches a user profile given a user's username **/
export function getUserProfile(username) {
    return request({
        url: API_BASE_URL + "/user/getUserProfile?username=" + username,
        method: 'GET'
    });
}

/** Fetches the petitions for a given user **/
export function getUserPetitions(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/user?username=" + username + "&page=" + page + "&size=" + size + "&sort=" + sort
        + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Fetches the most discussed petitions for a given user **/
export function getMostDiscussedPetitionsByUser(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitionsByUser?username=" + username + "&page=" + page +
            "&size=" + size + "&sort=" + sort
            + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Closes a petition given its details **/
export function closePetition(closePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/close",
        method: 'POST',
        body: JSON.stringify(closePetitionData)
    })
}

/** Creates a new argument by expert opinion given the new details **/
export function createArgumentByExpertOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Creates a new argument by popular opinion given the new details **/
export function createArgumentByPopularOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Fetches the arguments for a given petition **/
export function getArgumentsByRootPetitionId(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByRootPetitionId?rootPetitionId=" + petitionId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the arguments for a given argument **/
export function getArgumentsByArgumentId(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByArgumentId?argumentId=" + argumentId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the argument for a given ID **/
export function getArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/" + argumentId,
        method: 'GET'
    });
}

/** Fetches the parent argument for a given argument **/
export function getParentArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/getParentArgumentById?argumentId=" + argumentId,
        method: 'GET',
    });
}

/** Checks whether a petition is closed or not **/
export function isPetitionClosed(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/isPetitionClosed/" + petitionId,
        method: 'GET'
    })
}

/** Fetches the arguments for a given user **/
export function getUserArguments(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByUsernameAndAnonymity?username=" + username +
            "&showAnonymousContent=" + showAnonymousContent + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET'
    })
}

/** Fetches the number of supporting or opposing arguments for a given petition **/
export function countArgsByPetitionIdAndIsSupporting(petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countArgsByPetitionIdAndIsSupporting?petitionId=" + petitionId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Fetches the number of supporting or opposing arguments for a given argument **/
export function countArgsByArgumentIdAndIsSupporting(argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countAllByArgumentIdAndIsSupporting?argumentId=" + argumentId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Deletes an argument given its details **/
export function deleteArgument(deleteArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/delete",
        method: 'DELETE',
        body: JSON.stringify(deleteArgumentData)
    });
}

/** Updates an existing argument by popular opinion with new details **/
export function editArgumentByPopularOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Updates an existing argument by expert opinion with new details **/
export function editArgumentByExpertOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Fetches the most discussed petitions **/
export function getMostDiscussedPetitions(page, size, sort, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitions?page=" + page + "&size=" + size + "&sort=" + sort +
            "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments **/
export function getMostDiscussedArguments(pageSize, showClosed) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArguments?size=" + pageSize + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given petition **/
export function getMostDiscussedArgumentsForPetition(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForPetition?page=" + page + "&size=" + size + "&sort=" + sort +
            "&petitionId=" + petitionId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given argument **/
export function getMostDiscussedArgumentsForArgument(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForArgument?page=" + page + "&size=" + size + "&sort=" + sort +
            "&argumentId=" + argumentId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given user **/
export function getMostDiscussedArgumentsForUser(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForUser?page=" + page + "&size=" + size + "&sort=" + sort +
            "&username=" + username + "&showAnonymousContent=" + showAnonymousContent,
        method: 'GET'
    });
}

/** Fetches the data nodes required for the data visualisation of arguments **/
export function getDataNodes(contentId, isPetition) {
    return request({
        url: API_BASE_URL + "/arguments/getDataNodes?contentId=" + contentId + "&isPetition=" + isPetition,
        method: 'GET'
    });
}

/** Fetches data required for the petition report **/
export function getPetitionReport(petitionId) {
    return request({
        url: API_BASE_URL + "/arguments/getPetitionReport?petitionId=" + petitionId,
        method: 'GET'
    });
}

/** Searches for petitions given a query **/
export function searchPetitions(page, size, sort, showClosed, query) {
    return request({
        url: API_BASE_URL + "/petitions/search?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed
         + "&query=" + query,
        method: 'GET'
    });
}
