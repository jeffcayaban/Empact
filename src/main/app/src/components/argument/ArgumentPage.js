import React from 'react';

import { countArgsByArgumentIdAndIsSupporting, getArgumentById, getParentArgumentById, isPetitionClosed } from "../../utils/AppUtils";
import {formatTimestampToDate, getArgumentLabels} from "../../utils/Helpers";
import Loader from "../Loader";
import {Col, Grid, Panel, Row} from "react-bootstrap";
import {withRouter} from "react-router-dom";
import ExpertOpinion from "./content/ExpertOpinion";
import PopularOpinion from "./content/PopularOpinion";
import PreviewPetition from "./preview-context/PreviewPetition";
import PreviewArgument from "./preview-context/PreviewArgument";
import ShareContent from "../common/ShareContent";
import ArgumentActionPanel from "./ArgumentActionPanel";
import ArgumentVisualisation from "../common/argument-visualisation/ArgumentVisualisation";
import {ARGUMENT, EXPERT_OPINION, POPULAR_OPINION} from "../../constants";
import ArgumentPanel from "../common/ArgumentPanel";
import ContentRetrievalAlert from "../common/alerts/ContentRetrievalAlert";
import CreatorName from "../common/CreatorName";
import CreateArgumentPanel from "./CreateArgumentPanel";
import LastUpdated from "../common/LastUpdated";

/**
 * Represents the page for which the details of an argument is shown. It also displays any arguments that are made to
 * support or oppose the argument.
 */

class ArgumentPage extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            isLoading: true,
            argument: null,
            isArgumentCreator: false,
            showArgPreview: false,
            showPetitionPreview: false,
            isRootPetitionClosed: null,
            parentArgument: null,

            noOfForArguments: 0,
            noOfAgainstArguments: 0
        };

        this.renderArgumentContents = this.renderArgumentContents.bind(this);
        this.handleShowArgPreview = this.handleShowArgPreview.bind(this);
        this.handleShowPetitionPreview = this.handleShowPetitionPreview.bind(this);
    }

    // Fetches an argument from the server given an ID.
    getArgumentFromServer(argumentId) {
        getArgumentById(argumentId)
            .then(response => {
                this.getIsPetitionClosedFromServer(response.rootPetitionId);

                this.setState({
                    argument: response,
                    isLoading: false,
                    isArgumentCreator: this.props.currentUser && response.createdBy.username === this.props.currentUser.username
                })
            }).catch(error => this.setState({ isLoading: false }))
    }

    // Fetches the parent argument of a given argument's ID.
    getParentArgumentFromServer(argumentId) {
        getParentArgumentById(argumentId)
            .then(response => this.setState({ parentArgument: response }))
            .catch(error => this.setState({ parentArgument: null }))
    }

    // Fetches whether a petition is closed.
    getIsPetitionClosedFromServer(petitionId) {
        isPetitionClosed(petitionId)
            .then(response => this.setState({ isRootPetitionClosed: response }))
            .catch(error => console.error(error))
    }

    // Fetches the number of arguments that is made
    getNoOfArguments(argumentId, isSupporting) {
        countArgsByArgumentIdAndIsSupporting(argumentId, isSupporting)
            .then(response => {
                if (isSupporting) {
                    this.setState({ noOfForArguments: response })
                } else {
                    this.setState({ noOfAgainstArguments: response })
                }
            })
            .catch(error => console.error(error))
    }

    // Renders the argument contents based on its type.
    renderArgumentContents(argument, labels, includePreview) {
        const argType = argument.argumentType;

        if (argType === POPULAR_OPINION) {
            // The argument is by Popular Opinion.

            return <PopularOpinion
                argument={argument}
                includePreview={includePreview}
                labels={labels}
                showArg={this.handleShowArgPreview.bind(this)}
                showPetition={this.handleShowPetitionPreview.bind(this)}
                criticalQuestionId={argument.criticalQuestionId}
            />;
        } else if (argType === EXPERT_OPINION) {
            // The argument is by Expert Opinion.

            return <ExpertOpinion
                argument={argument}
                includePreview={includePreview}
                labels={labels}
                showArg={this.handleShowArgPreview.bind(this)}
                showPetition={this.handleShowPetitionPreview.bind(this)}
                criticalQuestionId={argument.criticalQuestionId}
            />;
        } else {
            // Display error message if the type is unidentifiable.

            return <Panel><Panel.Body>Unable to render the contents of the arguments</Panel.Body></Panel>
        }
    }

    // Updates the component's state to show the argument preview.
    handleShowArgPreview() {
        this.setState({ showArgPreview: !this.state.showArgPreview });
    }

    // Updates the component's state to show the petition preview.
    handleShowPetitionPreview() {
        this.setState({ showPetitionPreview: !this.state.showPetitionPreview });
    }

    // Generates a preview of either the parent argument or the root petition.
    generateContextPreview(labels) {
        const argument = this.state.argument;

        if (argument.parentArgumentId !== null && this.state.parentArgument !== null) {
            // If the parent argument exists then provide an option to preview it.

            return (
                <div>
                    <PreviewArgument
                        renderContents={this.renderArgumentContents}
                        labels={labels}
                        parentArgument={this.state.parentArgument}
                        show={this.state.showArgPreview}
                        handleHide={this.handleShowArgPreview}
                    />
                    <PreviewPetition
                        petitionId={argument.rootPetitionId}
                        show={this.state.showPetitionPreview}
                        handleHide={this.handleShowPetitionPreview}
                    />
                </div>
            );

        } else if (argument.rootPetitionId !== null) {
            // If the parent argument exists then provide an option to preview it.

            return <PreviewPetition petitionId={argument.rootPetitionId} show={this.state.showPetitionPreview}
                                    handleHide={this.handleShowPetitionPreview} />;
        }
    }

    // Provides the different options to modify an argument.
    static createArgumentActionPanel(argument, isRootPetitionClosed, isArgumentCreator) {
        if (!isRootPetitionClosed && isArgumentCreator) {
            // Assumes the petition is not closed and the user is the argument's creator.

            return (
                <ArgumentActionPanel
                    argumentId={argument.id}
                    isSupporting={argument.isSupporting}
                    isAnArgument={argument.parentArgumentId !== null}
                    parentSubjectId={argument.parentArgumentId === null ? argument.rootPetitionId : argument.parentArgumentId}
                />
            )
        }
    }

    componentDidMount() {
        const argumentId = this.props.match.params.id;
        this.getArgumentFromServer(argumentId);

        // Gets the parent argument (if an argument to another argument).
        this.getParentArgumentFromServer(argumentId);

        // Get the number of arguments made to the argument.
        this.getNoOfArguments(argumentId, true);
        this.getNoOfArguments(argumentId, false);
    }

    render() {
        if (!this.state.isLoading) {
            // Assumes that any data fetching has finished.

            const { argument, noOfForArguments, isRootPetitionClosed, parentArgument, isArgumentCreator,
                noOfAgainstArguments } = this.state;

            if (argument !== null && argument !== {}) {
                // Assumes that the argument exists.

                const labels = getArgumentLabels(argument, parentArgument, argument.agreesWithCQ);
                const creationDateTime = formatTimestampToDate(argument.creationDateTime);
                const argumentContents = this.renderArgumentContents(argument, labels, true);
                const argumentActionPanel = ArgumentPage.createArgumentActionPanel(argument, isRootPetitionClosed,
                    isArgumentCreator);

                return (
                    <div className={'secondaryPageContainerStyle'}>
                        { this.generateContextPreview(labels) }
                        <Grid className={"divNoPadding"} style={{ paddingTop: '30px' }} >
                            <Col md={12} lg={12}  className={"divNoPadding"}>
                                <div>
                                    <div className={"spaceBetweenElements"}>
                                        <div className={"argumentHeader"}>
                                            <h1 className={"argumentTitle"}>{labels[0]}</h1>
                                            <p>{labels[1]}</p>
                                        </div>
                                    </div>
                                </div>
                                <hr className={"argumentHr"} />
                                <div className={"spaceBetweenElements"}>
                                    <div className={"argumentCreationDetails"}>
                                        <span>
                                            {'By '}
                                            <CreatorName content={this.state.argument} />
                                            {` on ${creationDateTime}`}
                                        </span>
                                        <LastUpdated
                                            creationDateTime={argument.creationDateTime}
                                            lastUpdatedDateTime={argument.lastUpdatedDateTime} />
                                    </div>
                                    <ShareContent contentId={argument.id} contentType={ARGUMENT} />
                                </div>
                            </Col>
                        </Grid>
                        <Grid className={"divNoPadding"} style={{ paddingTop: '30px' }}>
                            <Col md={12} lg={12} className={'noSidePadding'}>
                                <div className={"spaceBetweenElements"} style={{ paddingBottom: '15px' }}>
                                    <div className={"contentIdNumber"}>{`ID: ${argument.id}`}</div>
                                    { argumentActionPanel }
                                </div>
                                { argumentContents }
                            </Col>
                        </Grid>
                        <CreateArgumentPanel argumentId={argument.id} isRootPetitionClosed={isRootPetitionClosed}
                            isArgumentCreator={isArgumentCreator} />
                        <Grid className={"divNoPadding"}>
                            <Col md={12} lg={12} className={'noSidePadding'}>
                                <ArgumentVisualisation contentId={argument.id} isPetition={false} />
                            </Col>
                        </Grid>
                        <Grid className={"divNoPadding"}>
                            <Row>
                                <Col md={6} lg={6}>
                                    <ArgumentPanel
                                        isSupporting={true}
                                        subjectId={argument.id}
                                        parentCreatedById={argument.createdBy.id}
                                        noOfArguments={noOfForArguments}
                                        isCreator={isArgumentCreator}
                                        parentContentType={ARGUMENT}
                                    />
                                </Col>
                                <Col md={6} lg={6}>
                                    <ArgumentPanel
                                        isSupporting={false}
                                        subjectId={argument.id}
                                        parentCreatedById={argument.createdBy.id}
                                        noOfArguments={noOfAgainstArguments}
                                        isCreator={isArgumentCreator}
                                        parentContentType={ARGUMENT}
                                    />
                                </Col>
                            </Row>
                        </Grid>
                    </div>
                );
            } else {
                // Will display an alert when the argument could not be fetched.
                return <div className={"alertContainer"}><ContentRetrievalAlert contentType={ARGUMENT} /></div>;
            }

        } else {
            // Will display a loader when data is being fetched.
            return <Loader isLoading={this.state.isLoading} />
        }
    }
}

export default withRouter(ArgumentPage);
