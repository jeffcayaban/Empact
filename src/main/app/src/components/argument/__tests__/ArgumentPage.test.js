import React from 'react';
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";
import * as AppUtils from "../../../utils/AppUtils";

import ArgumentPage from "../ArgumentPage";
import MockArgument from "../../../other/mock-data/MockArgument.js";

function createArgumentPage(props = {}) {
    const mockRouter = jest.fn();
    return <ArgumentPage.WrappedComponent {...props} params={{ router: mockRouter }} />;
}

describe('ArgumentPage', () => {

    it('should render successfully', () => {

        const testUsername = "testUsername";

        // Mock getArgumentById

        const originalGetArgumentById = AppUtils.getArgumentById;
        AppUtils.getArgumentById = (id) => Promise.resolve(MockArgument);

        // Mock getParentArgumentById

        const originalGetParentArgumentById = AppUtils.getParentArgumentById;
        AppUtils.getParentArgumentById = (id) => Promise.resolve(MockArgument);

        // Mock isPetitionClosed

        const originalIsPetitionClosed = AppUtils.isPetitionClosed;
        AppUtils.isPetitionClosed = (id) => Promise.resolve(true);

        // Mock countArgsByArgumentIdAndIsSupporting

        const originalCountArgs = AppUtils.countArgsByArgumentIdAndIsSupporting;
        AppUtils.countArgsByArgumentIdAndIsSupporting = (id, isSupporting) => Promise.resolve(5);

        const component = shallow(createArgumentPage({
            match: { params: { id: 'testId' } },
            currentUser: { username: testUsername }
        }));

        expect(toJson(component)).toMatchSnapshot();

        AppUtils.getArgumentById = originalGetArgumentById;
        AppUtils.getParentArgumentById = originalGetParentArgumentById;
        AppUtils.isPetitionClosed = originalIsPetitionClosed;
        AppUtils.countArgsByArgumentIdAndIsSupporting = originalCountArgs;
    });
});
