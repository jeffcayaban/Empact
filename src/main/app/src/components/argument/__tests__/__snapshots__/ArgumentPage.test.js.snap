// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArgumentPage should render successfully 1`] = `
<Loader
  isLoading={true}
/>
`;
